stdlib/LinearAlgebra/src/matmul.jl

```diff
@@ -1007,3 +1007,58 @@ function matmul3x3!(C::AbstractMatrix, tA, tB, A::AbstractMatrix, B::AbstractMat
     end # inbounds
     C
 end
+
+# Three-argument *
+*(A::AbstractMatrix, B::AbstractMatrix, x::AbstractVector) = _mat_mat_vec(A,B,x)
+
+*(α::Number, A::AbstractMatrix, x::AbstractVector) = _scalar_mat_vec(α,A,x)
+*(α::Number, A::AbstractMatrix, B::AbstractMatrix) = _scalar_mat_mat(α,A,B)
+*(A::AbstractMatrix, x::AbstractVector, α::Number) = _scalar_mat_vec(α,A,x)
+*(A::AbstractMatrix, B::AbstractMatrix, α::Number) = _scalar_mat_mat(α,A,B)
```

Comment:
Same here: changing the order is always dangerous.

Reply:
Indeed should be careful. Is 5-arg `mul!` specified to left-multiply with `α`?

Julia is happy to do this, BTW:
```
# Vector-Matrix multiplication
(*)(x::AdjointAbsVec,   A::AbstractMatrix) = (A'*x')'
(*)(x::TransposeAbsVec, A::AbstractMatrix) = transpose(transpose(A)*transpose(x))
```
Which is OK, if adjoint always reverses things. But my attempt to do something similar, for `_scalar_mat_mat -> Adjoint`, is dodgy:
```
_scalar_mat_mat(α, A::AdjointAbsVec, B) = _scalar_mat_vec(α', B', A')'
```

Reply:
I think I got all the orders straight now, on the assumption that `mul!` has the scalar on the right.